Add compiler hint for stdin().get_line() -> stdin().read_line()

### Code

I am learning Rust, and I keep typing `std::io::stdin().get_line()` instead of `std::io::stdin().read_line()` and the compiler doesn't give any useful hints.

### Current output
```
   Compiling playground v0.0.1 (/playground)
error[E0599]: no method named `get_line` found for struct `Stdin` in the current scope
 --> src/main.rs:5:13
  |
5 |     stdin().get_line(&mut string);
  |             ^^^^^^^^ method not found in `Stdin`
```
### Desired output
```
   Compiling playground v0.0.1 (/playground)
error[E0599]: no method named `get_line` found for struct `Stdin` in the current scope
 --> src/main.rs:5:13
  |
5 |     stdin().get_line(&mut string);
  |             ^^^^^^^^
  |
help: there is a method `read_line` with a similar name
  |
5 |     stdin().read_line(&mut string);
  |             ~~~~~~~~~
```
### Rationale and extra context

I keep accidentally making this mistake because I am a beginner.

### Other cases

_No response_

### Rust Version

rustc version 1.81.0 via https://play.rust-lang.org/?version=stable&mode=debug&edition=2021
I am not able to access my laptop right now, and it uses 1.80.1 anyway.

### Anything else?

_No response_

--- a/library/std/src/io/stdio.rs
+++ b/library/std/src/io/stdio.rs
@@ -394,6 +394,7 @@ impl Stdin {
     ///   in which case it will wait for the Enter key to be pressed before
     ///   continuing
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_confusables("get_line")]
     pub fn read_line(&self, buf: &mut String) -> io::Result<usize> {
         self.lock().read_line(buf)
     }

--- a/tests/ui/attributes/rustc_confusables_std_cases.rs
+++ b/tests/ui/attributes/rustc_confusables_std_cases.rs
@@ -23,4 +23,8 @@ fn main() {
     //~^ HELP you might have meant to use `push_str`
     String::new().append(""); //~ ERROR E0599
     //~^ HELP you might have meant to use `push_str`
+    let mut buffer = String::new();
+    let stdin = std::io::stdin();
+    stdin.get_line(&mut buffer).unwrap(); //~ ERROR E0599
+    //~^ HELP you might have meant to use `read_line`
 }

--- a/tests/ui/attributes/rustc_confusables_std_cases.stderr
+++ b/tests/ui/attributes/rustc_confusables_std_cases.stderr
@@ -106,7 +106,18 @@ help: you might have meant to use `push_str`
 LL |     String::new().push_str("");
    |                   ~~~~~~~~
 
-error: aborting due to 8 previous errors
+error[E0599]: no method named `get_line` found for struct `Stdin` in the current scope
+  --> $DIR/rustc_confusables_std_cases.rs:28:11
+   |
+LL |     stdin.get_line(&mut buffer).unwrap();
+   |           ^^^^^^^^ method not found in `Stdin`
+   |
+help: you might have meant to use `read_line`
+   |
+LL |     stdin.read_line(&mut buffer).unwrap();
+   |           ~~~~~~~~~
+
+error: aborting due to 9 previous errors
 
 Some errors have detailed explanations: E0308, E0599.
 For more information about an error, try `rustc --explain E0308`.

